target/linux/ipq40xx/image/mikrotik.mk

```diff
@@ -19,6 +19,15 @@ define Device/mikrotik_hap-ac2
 endef
 TARGET_DEVICES += mikrotik_hap-ac2
 
+define Device/mikrotik_ldf-5-ac
+	$(call Device/mikrotik_nor)
+	DEVICE_MODEL := LDF 5 ac (RBLDFG-5acD)
+	SOC := qcom-ipq4018
+	DEVICE_PACKAGES := ipq-wifi-mikrotik_ldf-5-ac -kmod-ath10k-ct \
+		kmod-ath10k-ct-smallbuffers rssileds -kmod-usb2 -kmod-usb3
```

Comment:
Is kmod-usb2 actually selected?

Reply:
@adschm
tnx; read the docs about the build system and all sort of options exist to add things, but not many to suppress things, other than to make a particular device not to appear at all in the menuconfig when something is set. Basically, as far as the kernel is concerned, for ipq40xx in config 5.4 get set:
CONFIG_PHY_QCOM_IPQ4019_USB=y
CONFIG_USB=y
CONFIG_USB_COMMON=y
CONFIG_USB_SUPPORT=y
All unnecessary for this device. The ipq40xx has USB Silicon in the SoC, but none of the peripheral hardware has been implemented in this device (it is for outdoor use). 
Those kernel configurations cannot be suppressed from from the device_packages description in mikrotik.mk, is my current understanding. But do correct me if I am wrong.
Image you compile for Mikrotik hAPac2, followed by a compilation for Mikrotik LDF ac. The latter firmware will have USB support though the hardware is not capable of it. If you know of a way to force compilation with 'CONFIG_USB_SUPPORT is not set', please advise. Obviously we can keep anything 'usb' out of mikrotik.mk. Perhaps a comment addition to the option 'LDF 5 ac' could advise to disable manually USB support. My only intention is to not have firmware elements created that serve no purpose.